pkg/tsdb/elasticsearch/client/ppl_request.go

```diff
@@ -0,0 +1,52 @@
+package es
+
+import (
+	"fmt"
+	"strings"
+)
+
+// PPLRequestBuilder represents a PPL request builder
+type PPLRequestBuilder struct {
+	index    string
```

Comment:
question, why the index is needed?

Reply:
index is needed to set the default PPL query `source = index` when initially building the request.
The PPLRequest struct which gets sent to ES only includes the Query field.